core/server/models/post.js

```diff
@@ -185,6 +189,52 @@ Post = ghostBookshelf.Model.extend({
         model.emitChange('deleted', Object.assign({usePreviousAttribute: true}, options));
     },
 
+    onDestroying: function onDestroyed(model) {
+        this.handleAttachedModels(model);
+    },
+
+    handleAttachedModels: function handleAttachedModels(model) {
+        model.related('tags').on('detaching', function onDetached(collection, tag) {
```

Comment:
1. why do we have to register two events? 🤔 

2. we should probably use `.once`, otherwise the event is not deregistered

Reply:
> why do we have to register two events?

We have to register both because 'detached' event doesn't contain the information about which objects were detached, that's why I'm closing over 'detaching' event first.

> we should probably use .once, otherwise the event is not deregistered. Or remove the event listener afterwards

Good point will try if `.once` does a trick, otherwise will deregister them after emitting the event.

Reply:
> We have to register both because 'detached' event doesn't contain the information about which objects were detached, that's why I'm closing over 'detaching' event first.

Interesting. Maybe leave a comment :)

Reply:
Sure, will do :)

Reply:
🤠